src/services/createRejection.ts

```diff
@@ -0,0 +1,10 @@
+import { genericRejection } from '@/components/rejection'
+import { Rejection } from '@/types/rejection'
+import { Component } from 'vue'
+
+export function createRejection<TType extends string>(type: TType, component?: Component): Rejection<TType> {
```

Comment:
Should we just make this an object rather than two positional arguments? Then it'll feel very similar to createRoute. And give us the ability to add options in the future without making a breaking change.